### PR TITLE
prime-factors: convert to use instance methods

### LINKED ISSUE #363: prime-factors: adjust API to use an instance method over class methods
As per #177, adjust the test suite to require a solution that uses an instance method instead of a class method.

_(see #344 as an example and the conversation about Function Object vs. Value Object)_

### PR DESCRIPTION
**Pull request by stkent**
Closes #363.

### CODE REVIEW

File: exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import org.junit.Test;
-import org.junit.Ignore;
```
Comment (stkent) [inline exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java]: Unused due to test suite parametrization.

File: exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java [hunk 2]
```diff
@@ -8,20 +7,22 @@
...
 
     @Parameters(name="Prime factors of {0} = {1}")
     public static Collection<Object[]> data() {
         return Arrays.asList(new Object[][]{
-                {1L, Arrays.asList()},
-                {2L, Arrays.asList(2L)},
-                {3L, Arrays.asList(3L)},
+                {1L, emptyList()},
+                {2L, singletonList(2L)},
+                {3L, singletonList(3L)},
```
Comment (stkent) [inline exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java]: All suggestions from IntelliJ.

Comment (FridaTveit) [inline exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java]: I'm more used to seeing this as `Collections.emptyList()` and `Collections.singletonList()` without the static import. That might make it more clear to someone who hasn't encountered them before?

Comment (stkent) [inline exercises/prime-factors/src/test/java/PrimeFactorsCalculatorTest.java]: 👍 sure, will revert!

**Pull request merged**